builders/reservations.go

```diff
@@ -0,0 +1,115 @@
+package builders
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/tfexplorer"
+	"github.com/threefoldtech/tfexplorer/client"
+	"github.com/threefoldtech/tfexplorer/models/generated/workloads"
+	wrklds "github.com/threefoldtech/tfexplorer/pkg/workloads"
+	"github.com/threefoldtech/tfexplorer/schema"
+)
+
+// ReservationClient is a client to deploy and delete reservations
```

Comment:
I would extract this client in it's own package and the builders in another, like so:
```
provision
├── builders
│   └── buiderXXX.go
│   └── ...
└── client.go
```
So the user of this lib would import the package with `github.com/threefoldtech/tfexplorer/provision`